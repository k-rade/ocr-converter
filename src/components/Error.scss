.error {
    margin: 1rem 0 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: 1rem;
    width: 100%;
    background-color: hsla(var(--color-hue), 36%, 54%, 0.25);
    border: 1px solid var(--color-primary-500);
    border-radius: 2px;
    padding: 0.5rem 1rem;

    &-message {
        color: var(--color-primary-900);
        display: flex;
        align-items: center;
        gap: 1rem;
        font-size: 0.875rem;
    }

    .icon-close {
        fill: var(--color-primary-900) !important;
        cursor: pointer;
    }

    .icon-warning-triangle>path {
        stroke: var(--color-primary-900);
    }
}