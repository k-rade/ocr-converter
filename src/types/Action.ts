export enum ActionType {
  SET_IMAGE_FILE,
  SET_IMAGE_URL,
  SET_LANGUAGE,
  SET_PROGRESS,
  SET_ERROR,
  CONVERT_IMAGE_INIT,
  CONVERT_IMAGE_SUCCESS,
  CONVERT_IMAGE_ERROR,
  INIT_NEW_CONVERSION,
}

export interface Action {
  type: ActionType;
  payload?: any;
}
