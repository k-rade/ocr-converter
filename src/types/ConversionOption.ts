export enum ConversionOption {
  FILE_UPLOAD,
  BY_LINK,
}
