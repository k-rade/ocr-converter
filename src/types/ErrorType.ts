export enum ErrorType {
  NO_ERROR,
  CONVERSION_FAILED,
  UNSUPPORTED_FILE_FORMAT,
  MAX_FILE_SIZE_EXCEEDED,
}
