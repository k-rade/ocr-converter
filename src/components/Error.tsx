import { ErrorType } from "../types/ErrorType";
import { FiX } from "react-icons/fi";

interface Props {
  onClose: () => void;
  error: ErrorType;
}

const Error = ({ onClose, error }: Props): JSX.Element => {
  const showErrorMessage = (): string => {
    switch (error) {
      case ErrorType.CONVERSION_FAILED:
        return "Image conversion failed. Please try again.";
      case ErrorType.UNSUPPORTED_FILE_FORMAT:
        return "Unsupported file format.";
      case ErrorType.MAX_FILE_SIZE_EXCEEDED:
        return "Maximum file size exceeded. Please use file up to 5 MB.";
      default:
        return "Image conversion failed. Please try again.";
    }
  };

  return (
    <div
      data-testid="error-banner"
      className="mb-6 border-l-[3px] border-red-500 p-4 bg-red-900/20"
    >
      <div className="flex items-center justify-between gap-4">
        <div className="font-medium text-red-400">Error</div>
        <FiX
          data-testid="error-icon-close"
          onClick={onClose}
          size="1.25rem"
          className="cursor-pointer stroke-white"
        />
      </div>
      <div className="text-white">
        <span>{showErrorMessage()}</span>
      </div>
    </div>
  );
};
export default Error;
