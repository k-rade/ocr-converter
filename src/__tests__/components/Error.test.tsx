import { fireEvent, getByRole, render } from "@testing-library/react";
import Error from "../../components/Error";
import { ErrorType } from "../../types/ErrorType";

describe("Error", (): void => {
  it("should call onClose", (): void => {
    const mockOnClose = jest.fn();
    const { getByTestId } = render(
      <Error onClose={mockOnClose} error={ErrorType.UNSUPPORTED_FILE_FORMAT} />
    );

    fireEvent.click(getByTestId("error-icon-close"));

    expect(mockOnClose).toHaveBeenCalled();
  });

  it("should initially be without error message", (): void => {
    const { baseElement } = render(
      <Error onClose={jest.fn} error={ErrorType.NO_ERROR} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render with unsupported file format message", (): void => {
    const { getByText } = render(
      <Error onClose={jest.fn} error={ErrorType.UNSUPPORTED_FILE_FORMAT} />
    );

    expect(getByText("Unsupported file format.")).toBeInTheDocument();
  });

  it("should render with conversion failed message", (): void => {
    const { getByText } = render(
      <Error onClose={jest.fn} error={ErrorType.CONVERSION_FAILED} />
    );

    expect(
      getByText("Whoops! Image conversion failed. Please try again.")
    ).toBeInTheDocument();
  });

  it("should render with maximum file upload size exceeded message", (): void => {
    const { getByText } = render(
      <Error onClose={jest.fn} error={ErrorType.MAX_FILE_SIZE_EXCEEDED} />
    );

    expect(
      getByText("Maximum file size exceeded! Please use file up to 5 MB.")
    ).toBeInTheDocument();
  });
});
