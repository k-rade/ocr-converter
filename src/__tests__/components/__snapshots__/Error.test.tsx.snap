// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should initially be without error message 1`] = `
<body>
  <div>
    <div
      class="mb-6 border-l-[3px] border-red-500 p-4 bg-red-900/20"
      data-testid="error-banner"
    >
      <div
        class="flex items-center justify-between gap-4"
      >
        <div
          class="font-medium text-red-400"
        >
          Error
        </div>
        <svg
          class="cursor-pointer stroke-white"
          data-testid="error-icon-close"
          fill="none"
          height="1.25rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </div>
      <div
        class="text-white"
      >
        <span>
          Image conversion failed. Please try again.
        </span>
      </div>
    </div>
  </div>
</body>
`;
